### PR TITLE
Make compatible with symfony security authenticator

### PR DESCRIPTION
**Pull request by Azoog**
Symfony Security has introduced a new security system in Symfony 5.3. In Symfony 6 the legacy system is removed in favor of this system.

With this breaking change I updated the bundle with this minimum requirements based on the last LTS:
- PHP 8.0
- Symfony 5.4

When enabled, all security systems of your project must use the new authenticator system. 
So I updated the sso-client in the same way in this PR: https://github.com/fulll/sso-client/pull/68

[v5 branch](https://github.com/fulll/security-bundle/tree/v5) for backward compatibility.

### CODE REVIEW

File: Controller/Annotations/ObfuscateRequest.php [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Rezzza\SecurityBundle\Controller\Annotations;
 
 /**
```
Comment (crozet-magenta) [inline Controller/Annotations/ObfuscateRequest.php:7]: can we declare this also as an attribute ?

Comment (Azoog) [reply Controller/Annotations/ObfuscateRequest.php:7]: It is not simple to replace the annotation to attribute system.
If necessary we should adapt it but I see no usage in our organisation.

Should we keep the `ObfuscateRequest` annotation?

Comment (crozet-magenta) [reply Controller/Annotations/ObfuscateRequest.php:7]: well if no one uses it, no need to spend time on it i think 🤷

File: composer.json [hunk 2]
```diff
@@ -10,17 +10,13 @@
         "homepage": "https://github.com/fulll/security-bundle/contributors"
     }],
     "require": {
-        "php": ">7.1.3",
-        "symfony/framework-bundle": "~2.6|~3.0|~4.0|~5.0",
-        "symfony/security-bundle" : "~2.6|~3.0|~4.0|~5.0",
+        "php": ">=8.0",
```
[the remark this answers is not in the document]
Comment (Azoog) [reply composer.json:13]: We can create a branch 5.X from the last tag so we could have some bug fixes for PHP 7.4 later.

**Review by crozet-magenta — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+on: push
+name: test
+
+jobs:
+  qa:
```
Comment (Kevin-koko) [inline .github/workflows/test.yml:5]: Can you add Phpstan here?

Comment (Kevin-koko) [reply .github/workflows/test.yml:5]: ahaha, it's better to make sure what we have now and be sure we won't add some bad things again
No need to do any clean

**Review by nio-dtp — APPROVED** [review]

**Review by truffo — APPROVED** [review]

**Review by crozet-magenta — APPROVED** [review]

**Pull request merged**